Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java

```diff
@@ -926,6 +926,9 @@ public void onBlockPistonRetract(BlockPistonRetractEvent event) {
 
     @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
     public void onBlockDispense(BlockDispenseEvent event) {
+        if (!this.plotAreaManager.hasPlotArea(event.getBlock().getWorld().getName())) {
+            return;
+        }
```

Comment:
I'm surprised we didn't have this here before. This will improve the speed of the server that has PlotSquared because it won't be doing a bunch of work and then checking if it's on a road or a plotworld since `location.isPlotRoad()` is an expensive call.